src/org/parosproxy/paros/security/CertData.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.parosproxy.paros.security;
+
+import org.bouncycastle.asn1.x509.GeneralName;
+
+import java.util.ArrayList;
+
+public class CertData {
+
+    private String commonName;
+    private ArrayList<GeneralName> subjectAlternativeNames;
+
+    public CertData() {
+        subjectAlternativeNames = new ArrayList<>();
+    }
+
+    public CertData(String commonName) {
+        this();
+        this.commonName = commonName;
+        if(commonName != null){
+            addSubjectAlternativeName(new GeneralName(GeneralName.dNSName, commonName));
+        }
+    }
+
+    public String getCommonName() {
+        return commonName;
+    }
+
+    public void addSubjectAlternativeName(GeneralName subjectAlternativeName){
+        subjectAlternativeNames.add(subjectAlternativeName);
+    }
+
+    public boolean isSubjectAlternativeNameIsCritical() {
+        return commonName == null;
+    }
+
+    public GeneralName[] getSubjectAlternativeNames() {
+        GeneralName[] subjectAlternativeNamesArray = new GeneralName[subjectAlternativeNames.size()];
+        return subjectAlternativeNames.toArray(subjectAlternativeNamesArray);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = commonName != null ? commonName.hashCode() : 0;
+        result = 31 * result + (subjectAlternativeNames != null ? subjectAlternativeNames.hashCode() : 0);
```

Comment:
This looks like magic, but it's unlikely to collide anyway

Reply:
Note that the hash code is not unique so collisions will happen sooner or later.